pkg/postage/service.go

```diff
@@ -162,22 +162,24 @@ func (ps *service) IssuerUsable(st *StampIssuer) bool {
 }
 
 // GetStampIssuer finds a stamp issuer by batch ID.
-func (ps *service) GetStampIssuer(batchID []byte) (*StampIssuer, error) {
+func (ps *service) GetStampIssuer(batchID []byte) (*StampIssuer, func(), error) {
 	ps.lock.Lock()
 	defer ps.lock.Unlock()
 	for _, st := range ps.issuers {
 		if bytes.Equal(batchID, st.data.BatchID) {
 			if !ps.IssuerUsable(st) {
-				return nil, ErrNotUsable
+				return nil, nil, ErrNotUsable
 			}
-			return st, nil
+			return st, func() { _ = ps.save() }, nil
```

Comment:
just for the context, why did we go from `func() error` -> `func()`?

Reply:
because the caller of the callback does not what to do with error, so we swallow the error here
in `service.Closer` we return the error thought